negy-gateway/src/main.rs

```diff
@@ -55,6 +58,14 @@ async fn fetch_nodes_unselected(node_pool_endpoint: &str) -> Result<Vec<NodeUnse
             name: n.name,
             version: n.version,
         })
+        .filter(|n| {
+           if let Some(min_version) = &args.min_version {
+                n.version.parse::<i32>().unwrap() >= min_version.parse::<i32>().unwrap()
```

Comment:
Don't parse as i32 but add `semver` crate and parse as Version. https://docs.rs/semver/latest/semver/